### DISCUSSION TITLE
Manual contract verification on Sepolia fails.

### DISCUSSION THREAD

**Discussion by SilviyaPBrayanova** [Cyfrin/foundry-full-course-cu#911] [Q&A]
Hi all,

I am trying to perform the manual contract verification as explained at TIMESTAMP ~8:06:50.

I get errors from the Etherscan compiler:
![compiler_output](https://github.com/Cyfrin/foundry-full-course-f23/assets/11355579/c22d3c25-7f2d-465e-bf95-5449dccd3ed0)
![bytecode_difference](https://github.com/Cyfrin/foundry-full-course-f23/assets/11355579/5f91b837-a436-4b65-81c7-7c380aac3cec)

I repeated the process a couple of times along with the video, trying to not miss a step or mess something up. Yet the error persists.
Any ideas?

**Comment by 0xusmanf** [comment]
Try it with Optimization Enabled to false.

**Comment by SilviyaPBrayanova** [comment]
Nope, that does not solve the problem. I played additionally and I noticed following behavior:

I. First scenario
1. I use this declaration in the contract **pragma solidity ^0.8.19;** (with the ^ symbol)
2. The forge builds the contract with compiler version 0.8.21 and then deploys on Sepolia testnet
3. On Sepolia`s Etherscan I specify compiler version 0.8.21 -> Verification FAILURE
4. On Sepolia`s Etherscan I specify compiler version 0.8.21 ->Verification  FAILURE

II Second scenario
1. I use this declaration in the contract **pragma solidity 0.8.19;** (exact version)
2. The forge builds the contract with compiler version 0.8.19 and then deploys on Sepolia testnet
3. On Sepolia`s Etherscan I specify compiler version 0.8.19 -> Verification SUCESS

Might matter for some reason.
10x

**Reply by wansidqi** [reply]
So basically, is it the compiler issue cause the verification failed?
